beacon_node/lighthouse_network/src/service/mod.rs

```diff
@@ -232,9 +232,9 @@ impl<AppReqId: ReqId, TSpec: EthSpec> Network<AppReqId, TSpec> {
                     SYNC_COMMITTEE_SUBNET_COUNT,
                     BLOB_SIDECAR_SUBNET_COUNT,
                 ),
-                max_subscribed_topics: 200,
+                max_subscribed_topics: 400,
```

Comment:
I'm confused as to how the peer in your logs is hitting the `max_subscribed_topics` limit.
So you received the new subscriptions 2 slots before the fork (I'm assuming transition from capella to deneb and you have already bumped `max_subscriptions_per_request`). 
An honest peer would be subscribed to at max 64 attestation + 4 sync committee + 7 core topic = 75 topics (before deneb, so not counting blob topics).

The error you are hitting is from this condition
https://github.com/libp2p/rust-libp2p/blob/da94dcb23b0d65a0e94965e6e8ddb14899053bd8/protocols/gossipsub/src/subscription_filter.rs#L152-L156
`currently_subscribed` should be 75 at max. `new_subscribed` can be 75 + 6 blob topics (with the new fork_digest). Assuming `unsubscribed = 0` in the code, we still aren't hitting the limit.
(81 + 75 = 156) which is less than both the limits.
So this is pointing to a bug in the code unless your setup somehow subscribes to more topics.

Reply:
Actually, I'm not sure how we hit the `max_subscriptions_per_request` limit either. The max number of topics we subscribe to at one go is equal to (75 + 6) < 160

Reply:
@qu0b can you expand a bit on the setup? are you changing config params that increase subnet counts or something similar?

Reply:
@pawanjay176 I am surprised that we are hitting this threshold as well. Will need to see how I can best debug this. Might need to build lighthouse myself and add some more metrics to output the subscribed topics. This is a bit time consuming, but I will get back to you

Reply:
I will look into the p2p library to see where this error originates. The setup we are running is with the testnet bootstrapper: https://github.com/antithesishq/ethereum-testnet-bootstrapper/tree/deneb the folder under builds/deneb-devnet-7

Reply:
@pawanjay176  We have a fork at epoch 1 (capella) and a fork at epoch 2 (deneb). So it might be that we are subscribing to topics from two different forks.

Reply:
After changing the genesis time from epoch 1 -> 5 for capella and epoch 2 -> 10 for deneb I no longer see the error occur.

Reply:
Shouldn't we also consider the two light client topics making it 9 core topics instead of 7?

Reply:
ah right, so nodes are subscribed to topics for 3 different forks! 

This shouldn't occur on mainnet but it's probably not uncommon on local testnets, and in theory a node could subscribe to a maximum of topics across 5 forks if network is configured that way, but that's not really a practical set up and something we need to be worried about.

Based on @pawanjay176's calculation above, it's unlikely we would hit the limit when Deneb fork occurs (81 + 75 = 156), however in the next fork we may need  (81 * 2 = 162), so it might be a good idea to bump the `max_subscriptions_per_request` while we're here. 

Also just thinking whether it's worth deriving the limit from constants & configurations, I think most are from constant except for `ATTESTATION_SUBNET_COUNT` which is configurable in network config, although we don't currently have any network using a value other than `64 `, so probably not a big deal either if we keep it simple and just hard code a limit here.

Reply:
@jimmygchen I don't really see the point in having 2x instead of for example 3x. The possible topics are whitelisted according the fork ids anyways and I don't see how this is a attack vector if we have 3x instead of 2x.

Reply:
@pawanjay176 @jimmygchen I made a change to the code that bases the `max_subscribed_topics` & `max_subscriptions_per_request` off of constants / configuration